### ISSUE TITLE
Bug with angular 1.6.1

### ISSUE THREAD

**Issue by BenevidesLecontes** [betsol/angular-input-modified#50]
After upgrading from angular 1.5.10 to angular 1.6.1, i noticed this error:
`TypeError: Cannot set property '$dirty' of null`, because this line
```
// Calling overloaded method.
originalSetPristine.apply(this, arguments);
```
So, i changed the line above to this and everything is working fine
```
if (this) {originalSetPristine.apply(this, arguments);}
```

**Comment by trimox** [comment]
Thanks!

**Comment by BenevidesLecontes** [comment]
@brunowego i'll do soon i can.

**Comment by slavafomin** [comment]
Hello! Thank you everyone for noticing this issue. I'm sorry I wasn't been able to fix it sooner.

It's fixed now in patch level immediate release`v2.3.4`. Feel free to try it out and let me know if it needs further attention.

Cheers!

**Comment by slavafomin** [comment]
Also, be advised, that solution provided by @BenevidesLecontes is incorrect and probably breaks some functionality. I would recommend to update as soon as possible if you were using it.
